### PR TITLE
refactor:Tidy up scalar module

### PR DESCRIPTION
**Pull request by winrhcp**
Rationale for this change
Mentioned in #234

What changes are included in this PR?
Move the Scalar trait to its own "scalar" file within base/scalar.

/claim #234

### CODE REVIEW

**Review by stuarttimwhite — CHANGES_REQUESTED** [review]
Just one request:
Can you change the file name to just scalar, not scalar_trait?

File: crates/proof-of-sql/src/base/mod.rs [hunk 1]
```diff
@@ -10,6 +10,7 @@ pub mod math;
 pub(crate) mod polynomial;
 pub(crate) mod proof;
 pub(crate) mod ref_into;
+/// TODO: add docs
```
Comment (JayWhite2357) [inline crates/proof-of-sql/src/base/mod.rs:13]:
```suggestion
/// This module contains the `Scalar` trait as well as the main, generic, implementations of it.
```

File: crates/proof-of-sql/src/base/scalar/mod.rs [hunk 2]
```diff
@@ -1,90 +1,16 @@
-//! This module contains the definition of the `Scalar` trait, which is used to represent the scalar field used in Proof of SQL.
+/// This module contains the definition of the `Scalar` trait, which is used to represent the scalar field used in Proof of SQL.
+pub mod scalar;
+pub use scalar::Scalar;
 mod error;
+pub use super::{encode::VarInt, ref_into::RefInto};
```
Comment (JayWhite2357) [inline crates/proof-of-sql/src/base/scalar/mod.rs:5]: Let's get rid of this.
```suggestion
```

**Review by JayWhite2357 — CHANGES_REQUESTED** [review]
Two very small requests on this, and I'll merge.

**Review by JayWhite2357 — APPROVED** [review]
Looks good. Thanks!

**Review by JayWhite2357 — APPROVED** [review]

**Pull request merged**